Userland/Libraries/LibCore/ArgsParser.cpp

```diff
@@ -586,8 +582,8 @@ void ArgsParser::add_option(Vector<DeprecatedString>& values, char const* help_s
         long_name,
         short_name,
         value_name,
-        [&values](StringView s) {
-            values.append(s);
+        [&values](StringView s) -> ErrorOr<bool> {
+            TRY(values.try_append(s));
```

Comment:
Similarly, an allocation failure here shouldn't crash the application, it should just gracefully error out.

Reply:
I must admit I'm not really sure what the difference is between `MUST(error)` (in ArgsParser.parse) and gracefully erroring out in this case?

It was my understanding that `MUST()` would produce a useful error output that would help the user investigate the error.

When I originally did this I had code that checked if the result of calling the `accept_value` function, and if it had an error it would warnln to the screen with that error, but then I realised there was the `MUST` macro and assumed that was more "idomatic" serenity c++.

I could definitely do with some further guidance on this :)

Reply:
MUST() will crash the program, giving you an assertion error.
gracefully erroring out means you can give the user an actual diagnostic.

Crashing because of an allocation failure is more acceptable than crashing because of invalid input of course, so if that's your cup of tea, something like this might be better suited here:
```c++
#define TRY_OR_ERROR_IF_NOT_OOM(expr, user_input) \
   ({ \
        auto&& _value = expr; \
        if (_value.is_error() && _value.error().is_errno() && _value.error().code() == ENOMEM) \
            return _value.release_error(); \
        if (_value.is_error()) {\
            warnln("Error while processing argument '{}': {}", user_input, _value.error()); \
            return false; \
        } \
       _value.release_value(); \
    })
```

Then here:
```suggestion
            TRY_OR_ERROR_IF_NOT_OOM(values.try_append(s), s);
```

Reply:
...which you can also use for the String cases.

Reply:
Ah, I quite like this! I think this is a neat approach to the problem and when testing it seems like a better user experience than what I was doing before.

Thanks for this suggestion!